File: September/Week3/Day4/Submissions/py/besttimetoSellStock.py
import math
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        pos = math.inf
        
        maxSellPrice = 0
        
        for i in prices:
            if i < pos:
                pos = i
            if i-pos > maxSellPrice:
                maxSellPrice = i-pos
                
        return maxSellPrice

File: September/Week3/Day4/Submissions/java/besttimetoSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        Integer minSellPrice = Integer.MAX_VALUE;
        int maxSellPrice = 0;
        
        for(int i=0;i<prices.length;i++) {
            if(prices[i] < minSellPrice) {
                minSellPrice = prices[i];
            }
            
            if(prices[i]-minSellPrice > maxSellPrice) {
                maxSellPrice = prices[i]-minSellPrice;
            }
        }
        
        return maxSellPrice;
    }
}